x/superfluid/keeper/migrate_test.go

```diff
@@ -1203,22 +1201,26 @@ const (
 // This test also asserts this same invariant at the very end, to ensure that all coins the accounts were funded with are accounted for.
 func (s *KeeperTestSuite) TestFunctional_VaryingPositions_Migrations() {
 	for i := 0; i < 10; i++ {
-		rand.Seed(time.Now().UnixNano() + int64(i))
+		r := rand.New(rand.NewSource(time.Now().UnixNano() + int64(i)))
 
 		// Generate random value from 0 to 50 for each position field
 		// This is how many positions of each type we will create and migrate
 		maxValue := 51
-		numBondedSuperfluid := rand.Intn(maxValue)
-		numUnbondingSuperfluidLocked := rand.Intn(maxValue)
-		numUnbondingSuperfluidUnlocking := rand.Intn(maxValue)
-		numVanillaLockLocked := rand.Intn(maxValue)
-		numVanillaLockUnlocking := rand.Intn(maxValue)
-		numNoLock := rand.Intn(maxValue)
+		numBondedSuperfluid := r.Intn(maxValue)
+		numUnbondingSuperfluidLocked := r.Intn(maxValue)
+		numUnbondingSuperfluidUnlocking := r.Intn(maxValue)
+		numVanillaLockLocked := r.Intn(maxValue)
+		numVanillaLockUnlocking := r.Intn(maxValue)
+		numNoLock := r.Intn(maxValue)
 
 		// Find the largest numPosition value and set numAccounts to be one greater than the largest position value
 		// The first account is used to create pools and the rest are used to create positions
 		largestPositionValue := osmoutils.Max(numBondedSuperfluid, numUnbondingSuperfluidLocked, numUnbondingSuperfluidUnlocking, numVanillaLockLocked, numVanillaLockUnlocking, numNoLock)
-		numAccounts := largestPositionValue.(int) + 1
+		largestPositionValue, ok := largestPositionValue.(int)
+		s.Require().True(ok)
+		numAccounts, ok := largestPositionValue.(int)
+		s.Require().True(ok, "largestPositionValue must be an int")
+		numAccounts++
```

Comment:
I think we can get away with this, why do we need to typecast `largestPositionValue` twice?
```suggestion
		largestPositionValue, ok := largestPositionValue.(int)
		s.Require().True(ok)
		numAccounts := largestPositionValue + 1
```

Reply:
hey @faddat! I was wrong here, your approach was actually the correct one. We needed to typecast twice because this line: 
```
largestPositionValue, ok := largestPositionValue.(int)
```
does not change the type of `largestPositionValue` (due to the way shadowing with typecasting in golang works). I pushed a [commit](https://github.com/osmosis-labs/osmosis/pull/6284/commits/611b7c03d3570f8bacf9648a3b24eff5edb1c592) to this branch that effectively reverts my suggested change to fix CI, sorry I was wrong